core/string.rbs

```diff
@@ -798,8 +798,7 @@ class String
   #
   # Related: String#concat, which takes multiple arguments.
   #
-  def <<: (Integer codepoint) -> self
-        | (string other_string) -> self
+  def <<: (string | Integer str_or_codepoint) -> self
```

Comment:
Combine the two overloads for the case passing a value of `String | Integer`.
(This can be improved with future version of Steep, but currently it doesn't work for this.)